### PR TITLE
feat(kernels): add opencl_batch_encoder for A770 batch processing

### PR DESCRIPTION
**Pull request by EffortlessSteven**
## Summary
Batch token encoding for Intel Arc A770 multi-sequence processing.

## Changes
- `TokenBatch` struct with token_ids, attention_masks, position_ids
- `PaddingStrategy` enum: MaxLength, Longest, Fixed, NoPadding
- `PackingStrategy` enum: Individual, Concatenated, SortedBins
- `BatchEncoder` for encoding variable-length sequences
- `DynamicBatcher` for length-similarity grouping
- `BatchMetrics` with padding waste tracking
- CPU reference implementations for all algorithms
- 49 tests including property tests

## Testing
```bash
cargo test -p bitnet-kernels --no-default-features --features cpu -- opencl_batch_encoder
```

### CODE REVIEW

**Comment by EffortlessSteven** [comment]
Closing: OpenCL scaffold  needs GPU hardware validation. Will be re-created when OpenCL CI lane is active.
